test/connect.tsx

```diff
@@ -0,0 +1,674 @@
+/// <reference path="../typings/main.d.ts" />
+
+import * as React from 'react';
+import * as chai from 'chai';
+import { mount } from 'enzyme';
+import { createStore } from 'redux';
+import { connect as ReactReduxConnect } from 'react-redux';
+// import { spy } from 'sinon';
+
+import {
+  GraphQLResult,
+  parse,
+  print,
+} from 'graphql';
+
+import ApolloClient from 'apollo-client';
+
+declare function require(name: string);
+import chaiEnzyme = require('chai-enzyme');
+
+chai.use(chaiEnzyme()); // Note the invocation at the end
+const { expect } = chai;
+
+import connect from '../src/connect';
+
+describe('connect', () => {
+
+  class Passthrough extends React.Component<any, any> {
+    render() {
+      return <span {...this.props} />;
+    }
+  };
+
+  class ProviderMock extends React.Component<any, any> {
+
+    static childContextTypes = {
+      store: React.PropTypes.object.isRequired,
+      client: React.PropTypes.object.isRequired,
+    };
+
+    getChildContext() {
+      return {
+        store: this.props.store,
+        client: this.props.client,
+      };
+    }
+
+    render() {
+      return React.Children.only(this.props.children);
+    }
+  };
+
+  describe('redux passthrough', () => {
+    it('should allow mapStateToProps', () => {
+      const store = createStore(() => ({
+        foo: 'bar',
+        baz: 42,
+        hello: 'world',
+      }));
+
+      const mapStateToProps = ({ foo, baz }) => ({ foo, baz });
+
+      @ReactReduxConnect(mapStateToProps)
```

Comment:
Whoa, what is this decorator magic?? I had no idea you could use react-redux in this way!